src/App/ComplexGeoData.cpp

```diff
@@ -184,6 +184,17 @@ bool ComplexGeoData::getCenterOfGravity(Base::Vector3d& unused) const
     return false;
 }
 
+std::optional<Base::Vector3d> ComplexGeoData::centerOfGravity() const

```

Comment:
Why adding another method to get the COG? Then just change the existing one to use std::optional.